lib/floe/workflow/choice_rule/boolean.rb

```diff
@@ -1,19 +0,0 @@
-# frozen_string_literal: true
-
-module Floe
-  class Workflow
-    class ChoiceRule
-      class Boolean < Floe::Workflow::ChoiceRule
-        def true?(context, input)
-          if payload.key?("Not")
-            !ChoiceRule.true?(payload["Not"], context, input)
-          elsif payload.key?("And")
-            payload["And"].all? { |choice| ChoiceRule.true?(choice, context, input) }
-          else
-            payload["Or"].any? { |choice| ChoiceRule.true?(choice, context, input) }
-          end
```

Comment:
originally, this called the static `ChoiceRule.true?`. So every run created an AST node on the fly.

The new code creates these all up front.

advantages/disadvantages of each.
My main issue here is the Data nodes were creating a node up front, while these were creating them on the fly.
Prefer to have all choice rules follow a similar approach is possible